File: Curve25519KeyGeneration/GenerateCurve25519Keypair.php
<?php

require_once ('Curve25519/Curve25519.php');

function generateCurve25519PrivateKey()
{
    return openssl_random_pseudo_bytes(32, $crypto_strong);
}

function calculateCurve25519PublicKey($secret)
{
    return (new Curve25519\Curve25519)->publicKey($secret);
}

function base64Encoding($input)
{
    return base64_encode($input);
}

echo 'Generate a key pair for Curve25519' . PHP_EOL;
// you need this library: https://github.com/lt/PHP-Curve25519

$privateKey = generateCurve25519PrivateKey();
$publicKey = calculateCurve25519PublicKey($privateKey);
echo 'base64 encoded key data:' . PHP_EOL;
echo 'Curve25519 PrivateKey: ' . base64Encoding($privateKey) . PHP_EOL;
echo 'Curve25519 PublicKey:  ' . base64Encoding($publicKey) . PHP_EOL;
?>

File: Curve25519KeyGeneration/GenerateCurve25519Keypair.cs
using System;
using System.Security.Cryptography;
using Elliptic;

class MainClass {
  public static void Main (string[] args) {

    Console.WriteLine("Generate a key pair for Curve25519");
    // you this library: https://github.com/hanswolff/curve25519

    byte[] privateKey = GenerateCurve25519PrivateKey();
    byte[] publicKey = CalculateCurve25519PublicKey(privateKey);
    Console.WriteLine("base64 encoded key data:");
    Console.WriteLine("Curve25519 PrivateKey: " + Base64Encoding(privateKey));
    Console.WriteLine("Curve25519 PublicKey:  " + Base64Encoding(publicKey));
  }

  static byte[] GenerateCurve25519PrivateKey() {
    RNGCryptoServiceProvider rngCsp = new RNGCryptoServiceProvider();
    byte[] key = new byte[32];
    rngCsp.GetBytes(key);
    return Curve25519.ClampPrivateKey(key);
  }

  static byte[] CalculateCurve25519PublicKey(byte[] secret) {
    return Curve25519.GetPublicKey(secret);
  }

  static string Base64Encoding(byte[] input) {
    return Convert.ToBase64String(input);
  }
}
